### PR TITLE
Work with dev httr2

### PR DESCRIPTION
**Pull request by hadley**
`req$policies$auth_oauth$cache$clear()` is an internal which you shouldn't really be calling, but it seems like something generally useful so I made an new function for y'all.

### CODE REVIEW

**Comment by jmaspons** [comment]
I'll do some tests with `set_osmapi_connection()` and check the httr2 changes before the merge, but looks good.

Thanks for the patch and for the socks 😊

File: R/osmapiR_connection.R [hunk 1]
```diff
@@ -158,9 +158,18 @@ authenticate_osmapi <- function() {
 #' @rdname authenticate_osmapiR
 #' @export
 logout_osmapi <- function() {
-  req <- httr2::request(base_url = get_osmapi_url())
-  req <- oauth_request(req)
-  req$policies$auth_oauth$cache$clear()
+
+  if (utils::packageVersion("httr2") >= "1.0.5.9000") {
+    getFromNamespace("oauth_cache_clear", "httr2")(
```
Comment (jmaspons) [inline R/osmapiR_connection.R:163]: I can't find the function `oauth_cache_clear()` in the httr2 repo, @hadley. Where is the commit/code?

Comment (hadley) [reply R/osmapiR_connection.R:163]: Oops, forgot to push it.

File: R/osmapiR_connection.R [hunk 2]
```diff
@@ -158,9 +158,18 @@ authenticate_osmapi <- function() {
 #' @rdname authenticate_osmapiR
 #' @export
 logout_osmapi <- function() {
-  req <- httr2::request(base_url = get_osmapi_url())
-  req <- oauth_request(req)
-  req$policies$auth_oauth$cache$clear()
+
+  if (utils::packageVersion("httr2") >= "1.0.5.9000") {
```
Comment (jmaspons) [inline R/osmapiR_connection.R:162]:
```suggestion
  if (utils::packageVersion("httr2") >= "1.0.6") {
```
Looking the recent history of httr2 that's the version we can rely on `oauth_cache_clear()`.

**Review by jmaspons — CHANGES_REQUESTED** [review]

File: R/osmapiR_connection.R (hunk 2 above)
Comment (hadley) [reply R/osmapiR_connection.R:162]: Now that it's released, you could just declare `(>= 1.0.6)` in the `DESCRIPTION` and drop all the optional code.

**Review by jmaspons — APPROVED** [review]
New commit depending on 1.0.6

**Pull request merged**